packages/service-client/odsp-client/src/odspClient.ts

```diff
@@ -0,0 +1,205 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { IDocumentServiceFactory } from "@fluidframework/driver-definitions";
+import {
+	OdspDocumentServiceFactory,
+	OdspDriverUrlResolver,
+	createOdspCreateContainerRequest,
+} from "@fluidframework/odsp-driver";
+import {
+	type ContainerSchema,
+	DOProviderContainerRuntimeFactory,
+	IFluidContainer,
+	FluidContainer,
+	IRootDataObject,
+} from "@fluidframework/fluid-static";
+import {
+	AttachState,
+	IContainer,
+	IFluidModuleWithDetails,
+} from "@fluidframework/container-definitions";
+import { IClient } from "@fluidframework/protocol-definitions";
+import { Loader } from "@fluidframework/container-loader";
+import { v4 as uuid } from "uuid";
+import {
+	IOdspResolvedUrl,
+	OdspResourceTokenFetchOptions,
+} from "@fluidframework/odsp-driver-definitions";
+import { ITokenResponse } from "@fluidframework/azure-client";
+// eslint-disable-next-line import/no-deprecated
+import { requestFluidObject } from "@fluidframework/runtime-utils";
+import { IRequest } from "@fluidframework/core-interfaces";
+import {
+	OdspClientProps,
+	OdspConnectionConfig,
+	OdspContainerServices,
+	OdspServiceAttributes,
+} from "./interfaces";
+import { OdspAudience } from "./odspAudience";
+
+/**
+ * OdspClient provides the ability to have a Fluid object backed by the ODSP service within the context of Microsoft 365 (M365) tenants.
+ *
+ * @alpha @sealed
+ */
+export class OdspClient {
+	private readonly documentServiceFactory: IDocumentServiceFactory;
+	private readonly urlResolver: OdspDriverUrlResolver;
+
+	public constructor(private readonly properties: OdspClientProps) {
+		const getSharePointToken = async (options: OdspResourceTokenFetchOptions) => {
+			const tokenResponse: ITokenResponse =
+				await this.properties.connection.tokenProvider.fetchStorageToken(
+					options.siteUrl,
+					"",
+				);
+			return {
+				token: tokenResponse.jwt,
+			};
+		};
+
+		const getPushServiceToken = async (options: OdspResourceTokenFetchOptions) => {
+			const tokenResponse: ITokenResponse =
+				await this.properties.connection.tokenProvider.fetchOrdererToken(options.siteUrl);
+			return {
+				token: tokenResponse.jwt,
+			};
+		};
+		this.documentServiceFactory = new OdspDocumentServiceFactory(
+			getSharePointToken,
+			getPushServiceToken,
+		);
+
+		this.urlResolver = new OdspDriverUrlResolver();
+	}
+
+	public async createContainer(containerSchema: ContainerSchema): Promise<{
+		container: IFluidContainer;
+		services: OdspContainerServices;
```

Comment:
Any good reason why services is not a member of IFluidContainer? Looking at getContainerServices(), it should be a static function, and the could be on any object.

Reply:
IFluidContainer consists of client-side properties and methods. @sashasimic @ChumpChief @Josmithr, can add more context.